.github/workflows/docs.yaml

```diff
@@ -28,6 +28,6 @@ jobs:
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
 
       - name: Check docs
-        run: make check-docs
+        run: make docs
```

Comment:
I'm a bit worried about this name change because it seems inconsistent. I think that `check-*` is more in line with our other make targets, for example the one below, which are designed to exit non-zero when something is wrong

Reply:
You are right, this does not fit nicely with other targets. I changed this to:
- Have `docs` target correctly generate and format the docs, including white noise handling
- Keep `check-docs` as a target to check against discrepancies and exiting with error if some are found